### PR TITLE
Refactor common.TestCase: speed up tests

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Refactor common.TestCase:
- Add testclass level holidays initialization for specified years.
- Add functions for checking if holiday with a specific name exist/not exist in specified years.
- Rework heavy tests (South Korea, Japan, Vietnam).

**Github CI tests total time:** 
| Python |  Beta  |   PR   |
|:------:|:------:|:------:|
|  3.10  | 90.14s | 40.63s |
|  3.11  | 89.61s | 51.21s |

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good (`make pre-commit`)
- [x] I've added tests to verify that the new code works and all tests pass locally (`make test`, `make tox`)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Review by arkid15r — COMMENTED** [review]
It seems that `assertHolidayNameInYears`/`assertNoHolidayNameInYears` could extend or replace existing `assertHolidayName`/`assertNoHolidayName` functionality.

Do you think it'd work for your use cases?

File: tests/countries/test_chile.py [hunk 1]
```diff
@@ -204,9 +200,9 @@ def test_national_unity(self):
             "2002-09-02",
         )
 
-        self.assertNoHolidayName(
+        self.assertNoHolidayNameInYears(
             "Día de la Unidad Nacional",
-            Chile(years=set(range(1915, 2050)).difference({1999, 2000, 2001})),
+            tuple(set(range(1915, 2050)).difference({1999, 2000, 2001})),
```
Comment (arkid15r) [inline tests/countries/test_chile.py:205]: Looks heavy.
I'd just add set - `{list, set, tuple}` - here https://github.com/dr-prodigy/python-holidays/blob/beta/tests/common.py#L82

File: tests/common.py [hunk 2]
```diff
@@ -81,7 +94,9 @@ def _parse_arguments(
             for item_arg in item_args:
                 if type(item_arg) in {list, tuple}:
                     items.extend(item_arg)
-                elif expand_items and isinstance(item_arg, Generator):
+                elif isinstance(item_arg, Generator) or isinstance(
+                    item_arg, range
+                ):
```
Comment (arkid15r) [inline tests/common.py:99]:
```suggestion
                elif isinstance(item_arg, (Generator, range)):
```

File: Makefile [hunk 3]
```diff
@@ -44,7 +44,7 @@ setup:
 
 test:
 	scripts/l10n/generate_mo_files.py
-	pytest --cov=. --cov-config=pyproject.toml --cov-report term --cov-report xml --durations 10 --durations-min=0.75 --no-cov-on-fail --numprocesses auto
+	pytest --cov=. --cov-config=pyproject.toml --cov-report term --cov-report xml --durations 10 --durations-min=0.75 --no-cov-on-fail --dist loadscope --numprocesses auto
```
Comment (KJhellico) [inline Makefile:47]:
```suggestion
	pytest --cov=. --cov-config=pyproject.toml --cov-report term --cov-report xml --durations 10 --durations-min=0.75 --dist loadscope --no-cov-on-fail --numprocesses auto
```

File: tests/common.py [hunk 4]
```diff
@@ -81,7 +94,9 @@ def _parse_arguments(
             for item_arg in item_args:
                 if type(item_arg) in {list, tuple}:
```
Comment (KJhellico) [inline tests/common.py:95]:
```suggestion
                if type(item_arg) in {list, set, tuple}:
```

File: tests/countries/test_chile.py (hunk 1 above)
Comment (KJhellico) [reply tests/countries/test_chile.py:204]: Done. Thank you!

File: tests/countries/test_chile.py [hunk 5]
```diff
@@ -224,12 +219,11 @@ def test_independence_holidays(self):
         self.assertHoliday(f"{year}-09-20" for year in range(1932, 1945))
 
     def test_columbus_day(self):
-        years = set(range(1922, 2000)).difference({1973})
+        years = tuple(set(range(1922, 2000)).difference({1973}))
```
Comment (arkid15r) [inline tests/countries/test_chile.py:222]: Can be removed now?
```suggestion
        years = set(range(1922, 2000)).difference({1973})
```

File: tests/countries/test_chile.py [hunk 6]
```diff
@@ -204,9 +199,9 @@ def test_national_unity(self):
             "2002-09-02",
         )
 
-        self.assertNoHolidayName(
+        self.assertNoHolidayNameInYears(
             "Día de la Unidad Nacional",
-            Chile(years=set(range(1915, 2050)).difference({1999, 2000, 2001})),
+            tuple(set(range(1915, 2050)).difference({1999, 2000, 2001})),
```
Comment (arkid15r) [inline tests/countries/test_chile.py:204]:
```suggestion
            set(range(1915, 2050)).difference({1999, 2000, 2001}),
```

File: Makefile [hunk 7]
```diff
@@ -17,7 +17,7 @@ check:
 	make test
 
 coverage:
-	pytest --cov=. --cov-config=pyproject.toml --cov-report term-missing --no-cov-on-fail --numprocesses auto
+	pytest --cov=. --cov-config=pyproject.toml --cov-report term-missing --no-cov-on-fail --dist loadscope --numprocesses auto
```
[the remark this answers is not in the document]
Comment (arkid15r) [reply Makefile]: I like it!

File: tests/countries/test_japan.py [hunk 8]
```diff
@@ -9,239 +9,223 @@
...
 from holidays.countries.japan import Japan, JP, JPN
 from tests.common import TestCase
 
 
 class TestJapan(TestCase):
     @classmethod
     def setUpClass(cls):
-        super().setUpClass(Japan)
+        years = range(1949, 2051)
+        super().setUpClass(Japan, years=years)
```
Comment (KJhellico) [inline tests/countries/test_japan.py:20]:
```suggestion
        super().setUpClass(Japan, years=range(1949, 2051))
```

**Review by arkid15r — COMMENTED** [review]
LGTM

Good stuff, thank you!

**Review by arkid15r — COMMENTED** [review]
LGTM

**Pull request merged**